jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -7740,9 +7747,16 @@ public String[] split(String regex, int max) {
 			} else {
 				int rLength = regex.lengthInternal();
 
-				char[] splitChars = regex.value;
+				char[] splitChars;
+				// if regex is compressed and this string isn't, decompress regex string
+				if (COMPACT_STRINGS && !compressed && (LATIN1 == regex.coder)) {
```

Comment:
This doesn't compile:
```
/home/keithc/space/openj9/jdk08/build/normal/jdk/j9jcl/jdk/src/share/classes/java/lang/String.java:3714: error: cannot find symbol
				if (COMPACT_STRINGS && !compressed && (LATIN1 == regex.coder)) {
				                                       ^
  symbol:   variable LATIN1
  location: class String
/home/keithc/space/openj9/jdk08/build/normal/jdk/j9jcl/jdk/src/share/classes/java/lang/String.java:3714: error: cannot find symbol
				if (COMPACT_STRINGS && !compressed && (LATIN1 == regex.coder)) {
				                                                      ^
  symbol:   variable coder
  location: variable regex of type String
```

Reply:
Replaced the check with `regex.isCompressed()` for both jdk 8 and 11 versions in https://github.com/eclipse-openj9/openj9/commit/b31226f895f3660b700971b8446627ffd16b0c0b